packages/calcite-components/src/utils/form.e2e.ts

```diff
@@ -0,0 +1,231 @@
+import { newE2EPage } from "@stencil/core/testing";
+import { html } from "../../support/formatting";
+import { componentsWithInputEvent } from "./form";
+
+describe("form", () => {
```

Comment:
Not all form-associated components have validation (e.g. slider, switch, meter, etc.), and IIRC you mentioned in the past that you don't like conditional tests/assertions. 

I'd also have to add new options for stuff like:
- the value to type to ensure the Change event fires (differs for input-date-picker, input-time-picker, input-number, etc.)
- A boolean for components like radio-button-group which need to have a child clicked rather than a value typed.

I'd be happy to make the change if you're okay with the conditional tests and additional options, just let me know.

Reply:
Alternatively I could make a validation common test helper or create some utils to reduce the repeat code in `form.e2e.ts`

Reply:
>  IIRC you mentioned in the past that you don't like conditional tests/assertions.

It's been a while, but maybe I brought it up in the context of individual, multipurpose tests? 🤔💭❓For the common test helpers, we do have conditional testing depending on what they are meant to cover (see [example](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/tests/commonTests.ts#L1637)). 

> I'd also have to add new options for stuff like:

Makes sense. We do have options for some of these specialized ones (see [example](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/tests/commonTests.ts#L682-L705)). 

> Alternatively I could make a validation common test helper or create some utils to reduce the repeat code in form.e2e.ts

It'd be great to have utils that we can use within the `formAssociated` test helper.